docs/api/mcp/index.mdx

```diff
@@ -0,0 +1,273 @@
+# Sourcegraph MCP Server
+
+<p className="subtitle">Connect AI agents and applications to your Sourcegraph instance's code search and analysis capabilities.</p>
+
+The Sourcegraph Model Context Protocol (MCP) Server provides AI agents and applications with programmatic access to your Sourcegraph instance's code search, navigation, and analysis capabilities through a standardized interface.
+
+<Callout type="info">MCP Server requires Sourcegraph 6.8+ and is available on all deployment types.</Callout>
+
+## Server Endpoint
+
+The MCP server is available at:
+
+```
+https://your-sourcegraph-instance.com/.api/mcp/v1
+```
+
+## Authentication
+
+The Sourcegraph MCP server supports three authentication methods:
+
+### Access Token (Query Parameter)
+
+Pass your access token as a query parameter:
+
+```
+https://your-sourcegraph-instance.com/.api/mcp/v1?token=YOUR_ACCESS_TOKEN
+```
+
+### Authorization Header
+
+Include your token in the Authorization header:
+
+```
+Authorization: token YOUR_ACCESS_TOKEN
+```
+
+### OAuth 2.0 Device Flow
+
+For programmatic access without storing credentials, use the OAuth 2.0 device flow:
+
+#### Prerequisites
+
+Before using the device flow, you must create an OAuth application in your Sourcegraph instance:
```

Comment:
We can [predefine oauth clients](https://sourcegraph.sourcegraph.com/-/editor?remote_url=github.com%2Fsourcegraph%2Fsourcegraph&branch=main&file=cmd%2Ffrontend%2Finternal%2Fauth%2Fidp%2Fdatabase%2Fpredefined_clients.go&editor=VSCode&version=2.4.0&start_row=12&start_col=0&end_row=20&end_col=0) as well, I wonder if we should do that.

@unknwon do you have thoughts here?

Then users would just need to copy the details here. No way to get a client secret then, so PKCE or RFC8628 device flow auth are required; which is probably what most clients should do anyways.

Reply:
My 2 cents: If we are aiming enterprises to use MCP feature to start with, let's ask users to create OAuth applications for now because our OAuth scopes are currently still all or nothing. Ideally, we could give away some less sensitive scopes OOTB for a predefined client, then if users want to grant more scopes, they can choose to create a different OAuth client for their MCP needs.